s3buildcache/src/main/kotlin/androidx/build/gradle/s3buildcache/S3StorageService.kt

```diff
@@ -127,6 +131,7 @@ class S3StorageService(
             return try {
                 val inputStream = client.getObject(request)
                 val blob = inputStream.response() ?: return null
+                if (blob.contentLength() == 0L) return null // return empty entries as a cache miss
```

Comment:
This change is a copy of [this](https://github.com/androidx/gcp-gradle-build-cache/pull/36) one